library/src/blas1/scal_host.hpp

```diff
@@ -0,0 +1,82 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#include "handle.h"
+#include "rocblas.h"
+#include "scal_device.hpp"
+#include "utility.h"
+
+template <typename T, typename U>
+rocblas_status rocblas_scal_batched_template(rocblas_handle handle,
+                                             rocblas_int    n,
+                                             const U*       alpha,
+                                             T*             x[],
+                                             rocblas_int    offsetx,
+                                             rocblas_int    incx,
+                                             rocblas_int    batch_count)
+{
+    static constexpr int NB = 256;
```

Comment:
Seems to be the common practice but I had a discussion with Scott and putting the block size as a template parameter might be better for future tuning.

Reply:
If blocksize is added, it should be the first, not the last, template parameter, so that later ones which are types can still be deduced automatically.

Reply:
Added blocksize as template parameter (with trtri as inspiration). (see 
9fe93bb)